### DISCUSSION TITLE
Welcome to Harbor Discussions!

### DISCUSSION THREAD

**Discussion by mehrancodes** [mehrancodes/laravel-harbor#61] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with Harbor.

**Comment by webard** [comment]
Hello,

Thank you for great work. I'm using it in my project to deploy PRs for preview.

In real-world there can be some difficulties with deployment. For example in my project there is no way to seed testing database. GitHub Action creates weekly artifact of truncated, anonymized version of database. Currently I have script in project that connect via API to GitHub, downloads artifact and import it to database. This needs to set api key for GitHub directly in project, but I think that job should be delegated to workflow. Unfortunately I don't know if Forge API let upload file to use in deployment script. 

In summary:
1. Harbor should have possibility to download database from other GitHub Action using artifacts
2. Harbor should upload database file to server
3. Deployment script should use database file and import it.

Best

**Reply by mehrancodes** [reply]
Hey Bartłomiej,

I glad you found Harbor useful. It should be possible to upload the database to the forge server by using the [execute command API](https://forge.laravel.com/api-documentation#execute-command) on Harbor side to create a sql file on the server and passing the database dump data as a string to it, something like: `echo "${{ env.DATABASE_DUMP }}" > path/to/storage/database.sql`

In this case you need to have the database artifact downloaded and stored as an env variable in the workflow before running the Harbor. Do you think this solution works for you?

**Reply by mehrancodes** [reply]
I think we can automate the database upload process but I need to give it a try first.

**Comment by vintagesucks** [comment]
Hi, thanks for making Harbor! It's something I've wanted for a long time. ❤️ 

---

I don't like making action workflows based on PR titles, and am currently experimenting with managing Harbor via PR labels like this:

```yaml
name: harbor

on:
  pull_request:
    types: [opened, reopened, synchronize, labeled, unlabeled, closed]

env:
  [...]

jobs:
  preview:
    runs-on: ubuntu-latest
    steps:
      - name: Setup PHP
        uses: shivammathur/setup-php@2.30.2
        with:
          php-version: '8.2'
          tools: mehrancodes/laravel-harbor
      - name: Start Provisioning
        if: |
          contains(github.event.pull_request.labels.*.name, 'preview')
          && contains(fromJson('["opened", "reopened", "synchronize", "labeled", "unlabeled"]'), github.event.action)
        run: harbor provision
      - name: Start Teardown
        if: |
          (
            contains(github.event.pull_request.labels.*.name, 'preview')
            && contains(fromJson('["closed"]'), github.event.action)
          ) ||
          (
            !contains(github.event.pull_request.labels.*.name, 'preview')
            && contains(fromJson('["labeled", "unlabeled"]'), github.event.action)
          )
        run: harbor teardown
```

Note that I also chose to use the [`shivammathur/setup-php`](https://github.com/shivammathur/setup-php) action instead of the `kirschbaumdevelopment/laravel-test-runner` image suggested in the docs.

---

While following the docs, I noticed that "[Provisioning Your Site](https://laravel-harbor.com/docs/provision/)" and "[Tearing Down The Site](https://laravel-harbor.com/docs/teardown/)" use different keys for the same secrets, which caused me to run into some errors when copying from the docs.

Examples:
* `secrets.FORGE_API_TOKEN` vs. `secrets.FORGE_TOKEN`
* `secrets.FORGE_SERVER_ID` vs. `secrets.FORGE_SERVER`

**Reply by mehrancodes** [reply]
Hi @vintagesucks! I'm glad you liked the work and appreciate the contributions! Using labels is a great idea - maybe we should find a place to add that to the documentation soon.

Is there any specific reason you used the `shivammathur/setup-php` image instead of the one suggested in the docs? I'll give it a try and update the documentation accordingly.

Thanks for pointing out the issue with the different keys. I'll update the docs for those as well.

**Comment by raimidev** [comment]
Hello, I'm trying to set up a daemon to start Horizon, is there any way to do that. I saw on the pipeline, that there's a pipe for Daemons, but I'm not sure how I should format it, and the documentation doesn't say anything about that. Thank you in advance.

**Reply by vintagesucks** [reply]
@raimidev I have never tried this myself, but is this what you are looking for?
- https://github.com/mehrancodes/laravel-harbor/pull/129
